core/src/ipld_ext.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+use cid::Cid;
+use serde::Serialize;
+
+/// IpldExt is a trait for serialization, deserialization, CID calculation assuming dagcbor.
+pub trait IpldExt: Serialize {
+    /// serde serialize to dag-json String.
+    fn to_json(&self) -> Result<String> {
+        let json_bytes = serde_ipld_dagjson::to_vec(self)?;
+        String::from_utf8(json_bytes).map_err(Into::into)
+    }
+
+    /// serde serialize to dag-cbor Vec u8
+    fn to_cbor(&self) -> Result<Vec<u8>> {
+        serde_ipld_dagcbor::to_vec(self).map_err(Into::into)
+    }
+
+    /// CID of serde serialize to dag-cbor
+    fn to_cid(&self) -> Result<Cid> {
+        let (cid, _) = self.to_dag_cbor_block()?;
+        Ok(cid)
+    }
+
+    /// serde serialize to dag-cbor and return the CID and bytes
+    fn to_dag_cbor_block(&self) -> Result<(Cid, Vec<u8>)> {
+        let cbor_bytes = self.to_cbor()?;
+        let hash = multihash_codetable::MultihashDigest::digest(
+            &multihash_codetable::Code::Sha2_256,
+            &cbor_bytes,
+        );
+        Ok((Cid::new_v1(0x71, hash), cbor_bytes))
+    }
+}
+
+impl<T: Serialize> IpldExt for T {}
```

Comment:
This is perfect, and demonstrates the value of a good trait. Now anyone who wants this behavior can import this trait and leverage its value.